### PR TITLE
Feat: N-to-1 Polynomial evaluation opening reduction

### LINKED ISSUE #138: If a NodeOutput poly has multiple openings, reduce openings to one
When traversing to a node in the prover, a NodeOutput polynomial can have multiple openings that need to be checked (virtually). We should reduce them to a single opening using the technique in PAZK 4.5.2.

### PR DESCRIPTION
**Pull request by AntoineF4C5**
This protocol will allow us to reduce N claims about different opening of the same polynomial to a unique claim at a new opening point.

This will likely require a Map of VirtualPolynomial -> MultilinearPolynomial, since we require the MLE to be materialized to compute h, the restriction of MLE to a line passing by the several opening points.
We could surely use the Trace to recover those polynomials.

--

Closes #138

### CODE REVIEW

**Review by Forpee — APPROVED** [review]
The core impl is solid. The N-to-1 reduction is correctly implemented with the right soundness checks (degree bound, boundary conditions, fiat-shamir binding of `h` before challenge derivation). The `Default/Custom` `ReductionFlow` is clean - allowing ops flexibility. 

Thanks for resolving #138 and un-ignoring those two soundness tests

File: jolt-atlas-core/src/onnx_proof/ops/sin.rs [hunk 1]
```diff
@@ -190,69 +169,176 @@ impl<F: JoltField, T: Transcript> OperatorProofTrait<F, T> for Sin {
...
         let rc_d = rc_encoding.one_hot_params().instruction_d;
-        let mut polys = vec![];
+        let mut polys = vec![CommittedPolynomial::TeleportNodeQuotient(node.idx)];
         polys.extend((0..sin_d).map(|i| CommittedPolynomial::SinRaD(node.idx, i)));
         polys.extend((0..rc_d).map(|i| CommittedPolynomial::TeleportRangeCheckRaD(node.idx, i)));
         polys
     }
 }
 
+fn prove_post_reduction_checks<F: JoltField, T: Transcript>(
```
[the remark this answers is not in the document]
Comment (AntoineF4C5) [reply jolt-atlas-core/src/onnx_proof/ops/sin.rs:222]: Yup indeed I had planned to rename so that it's consistent, then forgot !
